app/grandchallenge/algorithms/views.py

```diff
@@ -335,6 +338,67 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
+NOT_IN_REGISTRY_ERROR = "NOT_IN_REGISTRY"
+
+
+class AlgorithmImageMarkDesired(
+    LoginRequiredMixin,
+    ObjectPermissionRequiredMixin,
+    SuccessMessageMixin,
+    FormView,
+):
+    permission_required = "algorithms.change_algorithm"
+    raise_exception = True
+    form_class = ImageMarkDesiredForm
+    success_message = "Image successfully updated."
+    template_name = "algorithms/mark_desired_version.html"
+
+    @cached_property
+    def algorithm(self):
+        return self.algorithm_image.algorithm
+
+    @cached_property
+    def algorithm_image(self):
+        return get_object_or_404(AlgorithmImage, pk=self.kwargs["pk"])
```

Comment:
This property needs to go, you should get this from the validated data from the form.